src/coreclr/vm/threadsuspend.cpp

```diff
@@ -2479,8 +2496,10 @@ void RedirectedThreadFrame::ExceptionUnwind()
     Thread* pThread = GetThread();
 
     // Allow future use to avoid repeatedly new'ing
-    pThread->UnmarkRedirectContextInUse(m_Regs);
-    m_Regs = NULL;
+    if (pThread->UnmarkRedirectContextInUse(m_Regs))
+    {
```

Comment:
Here - can we just check `UseContextBasedThreadRedirection()` and if not, then not call Unmark and not assign null?